#153009251 setup unit testing for the front end

fuchodeveloper opened the pull request:
#### What does this PR do?

Setup unit testing on client side

#### Description of Task to be completed?

- Setup dependencies necessary for testing client side
- Run karma init to create base karma config
- Create spec folder to hold client tests
- Create dummy test to make sure test works
- To run client test `npm run test:client`

#### How should this be manually tested?

- Clone app `git clone https://github.com/andela/k2-cfh.git`
- Run cd k2-cfh to access app folder
- Run `npm install` to install app dependencies
- Run `npm install -g karma` to install karma (use `sudo npm install -g karma` if that fails)
- Run client test `npm run test:client`

#### Any background context you want to provide?

- Ensure Google Chrome is installed on your machine

#### What are the relevant pivotal tracker stories?

153009251

#### Screenshots (if appropriate)

<img width="1440" alt="screen shot 2017-11-22 at 9 51 33 pm" src="https://user-images.githubusercontent.com/22713293/33150242-e3d2bf30-cfd2-11e7-8bcc-7459665d372d.png">

<img width="1439" alt="screen shot 2017-11-22 at 9 49 38 pm" src="https://user-images.githubusercontent.com/22713293/33150259-f2f5559a-cfd2-11e7-82d0-2519db9384ba.png">

#### Questions:
N/A

damakahindi commented on .gitignore:
```diff
@@ -5,4 +5,3 @@ public/lib
 c4h_keys.txt
 .env
 package-lock.json
-
```
Ensure not to push any empty lines at the end of files that no other changes you made
.

damakahindi commented on .eslintignore:
```diff
@@ -0,0 +1,2 @@
+spec
+app/models
```
Please do not ignore the models. There is a way to disable eslint on a specific line or rule.

damakahindi commented on .hound.yml:
```diff
@@ -1,3 +1,5 @@
-  config_file: .eslintrc
\ No newline at end of file
+  config_file: .eslintrc
+  ignore_file: .eslintignore
```
Nice work on adding the .eslintignore

damakahindi commented:
Fix those things and I will merge this PR. Good work.

damakahindi commented on .env:
```diff
@@ -0,0 +1 @@
+MONGOHQ_URL=mongodb://localhost:27017/k2-cfh
```
Remove this file.

Commit:
chore(frontend-testing): setup unit testing for the front end
fix merge conflict in files
ignore eslint on model files
fix identation and eslint in article.js model file
Finishes #153009251

Commit:
chore(frontend-testing): setup unit testing for the front end
fix merge conflict in files
ignore eslint on model files
fix identation and eslint in article.js model file
Finishes #153009251
